crypto/ui/ui_lib.c

```diff
@@ -857,47 +888,51 @@ int UI_set_result(UI *ui, UI_STRING *uis, const char *result)
             BIO_snprintf(number2, sizeof(number2), "%d",
                          uis->_.string_data.result_maxsize);
 
-            if (l < uis->_.string_data.result_minsize) {
+            if (len < uis->_.string_data.result_minsize) {
                 ui->flags |= UI_FLAG_REDOABLE;
-                UIerr(UI_F_UI_SET_RESULT, UI_R_RESULT_TOO_SMALL);
+                UIerr(UI_F_UI_SET_RESULT_EX, UI_R_RESULT_TOO_SMALL);
                 ERR_add_error_data(5, "You must type in ",
                                    number1, " to ", number2, " characters");
                 return -1;
             }
-            if (l > uis->_.string_data.result_maxsize) {
+            if (len > uis->_.string_data.result_maxsize) {
                 ui->flags |= UI_FLAG_REDOABLE;
-                UIerr(UI_F_UI_SET_RESULT, UI_R_RESULT_TOO_LARGE);
+                UIerr(UI_F_UI_SET_RESULT_EX, UI_R_RESULT_TOO_LARGE);
                 ERR_add_error_data(5, "You must type in ",
                                    number1, " to ", number2, " characters");
                 return -1;
             }
         }
 
         if (uis->result_buf == NULL) {
-            UIerr(UI_F_UI_SET_RESULT, UI_R_NO_RESULT_BUFFER);
+            UIerr(UI_F_UI_SET_RESULT_EX, UI_R_NO_RESULT_BUFFER);
             return -1;
         }
 
-        OPENSSL_strlcpy(uis->result_buf, result,
-                    uis->_.string_data.result_maxsize + 1);
+        memcpy(uis->result_buf, result, len);
+        if (len < uis->_.string_data.result_maxsize)
+            uis->result_buf[len] = '\0';
```

Comment:
I think this should be ```<=``` so that the NUL terminator is still present for the equality case, as would be ensured by the strlcpy that is being removed.

Reply:
No.  The result buffer is externally provided, and might be exactly maxsize bytes long.  I know that's not what we usually mean with a string in C, and I cannot remember exactly why I designed the UI API this way, sorry...

Reply:
I was looking at documentation that said "maxsize (not including trailing NUL byte)",  which to me implied that maxsize+1 was needed, but I won't push it.

Reply:
Well, maybe I will just ask a clarifying question that you are claiming that the previous code was buggy?  I do note that the result_maxsize+1 appears in several places already.